modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+
+=== Limit Results
+
+Mule runtime allows the connector to handle streaming gracefully. However, that does not mean that it's a good idea to move large chunks of data from the database to Mule. Even with streaming, a typical SQL query can return many rows, each one containing a lot of information. This generates:
+
+* Large network overhead
+* A large resources footprint on the database server
+* Large memory and requirements in the Mule Runtime
+
+The Select operation provides parameters (`fetchSize` and `maxRows`) to help with this:
+
+[source,xml,linenums]
+----
+<db:select fetchSize="200" maxRows="1000" config-ref="dbConfig">
+  <db:sql>select * from some_table</db:sql>
+</db:select>
+----
+
+This example tells the connector not to fetch more that `1000` total rows as the result of an SQL `SELECT` query, but to try to only stream results from the database to the Database connector `200` at a time. So in this example, it would require 5 separate network round trips to retrieve 1000 matching rows.
+
+The combination limits the total amount of information that will be retrieved (the `maxRows` value) and guarantees that the data is returned from the database over the network in smaller chunks (the `fetchSize` value).
+
+[NOTE]
+The `fetchSize` is a hint to the JDBC driver and need not always be enforced. The behavior varies between JDBC driver providers. Often, the default `fetchSize` for a JDBC driver is `10`.
+
+
+== Query Timeout Reference
+
+Sometimes database queries take a long time to execute. The following factors often cause delays in query execution:
+
+* An inefficient query, such as one having improper indexing that iterates over many rows
+* A busy RDBMS or network
+* A lock contention
+
+Generally, it's recommended to set a timeout on the query. To manage timeouts, configure `queryTimeout` and `queryTimeoutUnit`. The following example shows how to set a timeout for the Select operation, but all operations support setting a timeout:
+
+[source,xml,linenums]
+----
+<db:select queryTimeout="0" queryTimeoutUnit="SECONDS" config-ref="dbConfig">
+   <db:sql>select * from some_table</db:sql>
+</db:select>
+----
+
+= Examples: Insert, Update, and Delete Data with the Database Connector - Mule 4
+
+The Database connector supports insert, update, and delete operations.
+Similar to the select operation, these operations support dynamic queries and
+parameterization by using embedded DataWeave transformations while also
+supporting `fetchSize`, `maxRows`, and timeout parameters. Although examples of input
+parameterization, dynamic queries, and parameter usage are provided, you can consult
+the xref:db/db-connector-query.adoc[SELECT operation documentation]
+for more information.
+
+== Insert Example
+
+Suppose there is a database schema named *Products* that has a table named
```

Comment:
I thought you were going to rewrite this?

Reply:
This is the rewrite. Removed xref in the meantime.